pyrit/orchestrator/fuzzer_orchestrator.py

```diff
@@ -0,0 +1,427 @@
+            try: 
+                target_seed_obj = await self._apply_template_converter(current_seed._template)
+            except:
+                raise ValueError("Invalid template")
+                
+            target_template = PromptTemplate(target_seed_obj.output_text,parameters = ["prompt"])
+
+            # convert the target_template into a prompt_node to maintain the tree information
+            target_template_node = PromptNode(template = target_seed_obj, parent= current_seed) 
+
+            #3. Append prompt converted template with prompt. Apply each of the prompts (questions) to the template. 
+            
+            jailbreak_prompts = []
+            for prompt in self._prompts:
+                jailbreak_prompts.append(
+                    target_template.apply_custom_metaprompt_parameters(prompt=prompt) 
+                )
+
+            #4. Apply prompt converter if any and send request to a target 
+            
+            requests: list[NormalizerRequest] = []
+            for jailbreak_prompt in jailbreak_prompts:
+                requests.append(
+                    self._create_normalizer_request(
+                    prompt_text=jailbreak_prompt,
+                    prompt_type="text",
+                    converters=self._prompt_converters,
+                    )
+                )
+
+            for request in requests:
+                request.validate()
+
+            responses: list[PromptRequestResponse]
+            responses = (   
+            await self._prompt_normalizer.send_prompt_batch_to_target_async(
+                requests=requests,
+                target=self._prompt_target,
+                labels=self._global_memory_labels,
+                orchestrator_identifier=self.get_identifier(),
+                batch_size=self._batch_size, 
+                )
+            )
+
+            response_pieces = [response.request_pieces[0] for response in responses]
+            
+            #5. Apply scorer on the response and based on the scorer return if jailbreak successful or not.
+
+            scored_response = [] 
+            for response in response_pieces:
+                scored_response.append(
+                    self._scorer.score_async(response))
+            
+            batch_scored_response = await asyncio.gather(*scored_response)
+
+            score_values = [score[0].get_value() for score in batch_scored_response]
+ 
+            #6. Update the rewards for each of the node.
+           # self._num_jailbreak = sum(score_values)
+            self._num_jailbreak = score_values.count(True)
+            num_executed_queries = len(score_values)
+
+            self._current_jailbreak += self._num_jailbreak
+            self._current_query += num_executed_queries
+
+            new_nodes = []
+            if self._num_jailbreak > 0: #successful jailbreak
+                # append the template to the initial template list.

```

Comment:
```suggestion
                # The template resulted in at least one jailbreak so it will be part of the results
                # and a potential starting template for future iterations.

```
This comment is almost identical to what the code says and doesn't help much, but let's make it useful, perhaps like this?